src/OrchardCore.Modules/OrchardCore.OpenId/Drivers/OpenIdClientSettingsDisplayDriver.cs

```diff
@@ -0,0 +1,94 @@
+            INotifier notifier,
+            IHtmlLocalizer<OpenIdConnectSettingsDisplayDriver> stringLocalizer,
+            IMemoryCache memoryCache,
+            IOpenIdClientService openIdConnectService
+            )
+        {
+            _httpContextAccessor = httpContextAccessor;
+            _notifier = notifier;
+            T = stringLocalizer;
+            _memoryCache = memoryCache;
+            _openIdConnectService = openIdConnectService;
+        }
+
+        public override IDisplayResult Edit(OpenIdClientSettings settings, BuildEditorContext context)
+        {
+            if (context.GroupId == SettingsGroupId && _memoryCache.Get(RestartPendingCacheKey) != null)
+                _notifier.Warning(T["The site needs to be restarted for the settings to take effect"]);
+
+            var requestUrl = _httpContextAccessor.HttpContext.Request.GetDisplayUrl();
+            return Shape<OpenIdClientSettingsViewModel>("OpenIdClientSettings_Edit", model =>
+            {
+                model.DisplayName = settings.DisplayName;
+                model.TestingModeEnabled = settings.TestingModeEnabled;
+                model.AllowedScopes = settings.AllowedScopes != null ? string.Join(",", settings.AllowedScopes) : null;
+                model.Authority = settings.Authority;
+                model.CallbackPath = settings.CallbackPath;
+                model.ClientId = settings.ClientId;
+                model.ClientSecret = _openIdConnectService.Unprotect(settings.ClientSecret);
+                model.SignedOutCallbackPath = settings.SignedOutCallbackPath;
+                model.SignedOutRedirectUri = settings.SignedOutRedirectUri;
+            }).Location("Content:2").OnGroup(SettingsGroupId);
+        }
+
+        public override async Task<IDisplayResult> UpdateAsync(OpenIdClientSettings settings, IUpdateModel updater, string groupId)
+        {
+            if (groupId == SettingsGroupId)
+            {
+                var previousClientSecret = settings.ClientSecret;
+                var model = new OpenIdClientSettingsViewModel();
+                await updater.TryUpdateModelAsync(model, Prefix);
+                settings.DisplayName = model.DisplayName;
+                settings.TestingModeEnabled = model.TestingModeEnabled;
+                settings.AllowedScopes = model.AllowedScopes.Split(new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+                settings.Authority = model.Authority;
+                settings.CallbackPath = new PathString(model.CallbackPath);
+                settings.ClientId = model.ClientId;
+                settings.ClientSecret = _openIdConnectService.Protect(model.ClientSecret);
+                settings.SignedOutCallbackPath = model.SignedOutCallbackPath;
+                settings.SignedOutRedirectUri = model.SignedOutRedirectUri;
+
+                if (_openIdConnectService.IsValidOpenIdConnectSettings(settings, updater.ModelState) && _memoryCache.Get(RestartPendingCacheKey) == null)
```

Comment:
Same comment as in previous PR, you can just restart the site when the form is updated, as long as there is a warning in the default view that saving the settings will recycle the current tenant.

Reply:
I suggest we do that separately as we'll want to update the server and validation features to do the same thing.

Reply:
https://github.com/OrchardCMS/OrchardCore/issues/462 tracks that.